OHHTTPStubs/Sources/OHHTTPStubs.m

```diff
@@ -442,13 +442,23 @@ - (void)startLoading
                         case 301:
                         case 302:
                         case 307:
-                        case 308:
+                        case 308: {
                             //Preserve the original request method and body, and set the new location URL
                             mReq = [self.request mutableCopy];
                             [mReq setURL:redirectLocationURL];
+
+                            // Drop certain headers in accordance with
+                            // https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1411532-httpadditionalheaders
+                            [mReq setValue:nil forHTTPHeaderField:@"Authorization"];
+                            [mReq setValue:nil forHTTPHeaderField:@"Connection"];
+                            [mReq setValue:nil forHTTPHeaderField:@"Host"];
+                            [mReq setValue:nil forHTTPHeaderField:@"Proxy-Authenticate"];
+                            [mReq setValue:nil forHTTPHeaderField:@"Proxy-Authorization"];
+                            [mReq setValue:nil forHTTPHeaderField:@"WWW-Authenticate"];
```

Comment:
This could maybe be a `for header in @[@"…",@"…",…]` loop instead?

Reply:
I pushed a commit to put this into it's own method since where it is now is already a bit large.